### ISSUE TITLE
Less-painful breaches concept

### ISSUE THREAD

**Issue by Terror4000rus** [HaloSpaceStation/HaloSpaceStation13#2749]
tl;dr - an object, which doesn't allow atmosphere bypass it. Place under hull walls, windows and airlocks.

It's a concept, which should make combat not so frustrating in specific circumstances.
...Mainly, I talk about breaches in space objects.
Usual situation - your space object gets breaches during space battles or infiltration with wia windows or walls.

It causes problems with atmosphere, which leads to:
* falling in mid-combat due stabilization of pressure between rooms;
* necessity in wearing EVA gear (suit and oxygen);
* impossibility of healing due air-tight suits (they haven't injection ports, for now - you hate to remove a part of armor to use syringe, or use only auto-injectors), which stimulates people to ghost in case of paincrit;
* openning closed firedoors.
This is a list of first things that come to my mind. I think some unsatisfied with currect EVA gameplay players could say more.
In general, it is just frustrating.

I suggest:
* make floor indestructible (at least one which leads to space),
* make a layer of hull walls, which in case of breaking will activate some-high-tech system, which ~~(which,which,which, my english is too poor to find other word ><)~~ will not allow atmosphere pass its tile, but will allow characters to do so.
* add the system to all hull windows.

What I expect:
* EVA infiltration still exists, but it doesn't causes atmospheric problems,
* more possibilities for medics to realize their role at space objecties,
* more variety in armor sets.

The system (actually, just an object):
Icon - I don't know, maybe a small vent (like one in TG airlocks).
Functionality - doesn't allow air bypass it. Nothing more. Indestructable.
Placements - the way warries between manual placement (in airlocks and tiles with walls) and automatic spawning at tiles after breaking its wall.
Can be easily added to the game - a little code (can write, if needed).

~~Wanna hope you will understand what I wanna say.~~

**Comment by bloxgate** [comment]
We do have emergency firelocks that can be put in hallways/over doors/on windows. So long as there's an air alarm in the same area as the firelock, it should trigger them to close. You'll still have problems for people in the affected area, but it will stop people from just opening a door to vacuum.

I was under the impression that these were mapped in, are they not present/not working?

**Comment by ghost** [comment]
Emergency firelocks are mapped in and work normally, they're on windows and doorways so whenever a breach happens it's limited to a (relatively) small area.

Inevitably though, the entire ODP will be breached through assaulters maliciously opening doors (for combat advantage), creating new breaches (both for flanks and aforementioned combat advantage, there's a limited amount of EVA armour on the ODP) or both sides being forced to move through (thereby spreading the breach to other rooms) breached areas to reach an objective/threat/etc.

**Comment by Terror4000rus** [comment]
> We do have emergency firelocks that can be put in hallways/over doors/on windows. So long as there's an air alarm in the same area as the firelock, it should trigger them to close. You'll still have problems for people in the affected area, but it will stop people from just opening a door to vacuum.

Nope. Every round with breaching in ODP from same is the same - covenant removes a window, cause low atmosphere and keeps it forever open - either by going through the firedoor constantly, or just by welding it. You don't need atmosphere in the ship since you attack it from space. And it's annoying.
...And I even don't speek about poor ODP mapping. Not every room has air alarm. Pump at first deck at empty and vents work too slow. You just can't fix atmosphere in 5 minutes - especially, when you enemy doesn't want so and attack the station.

> I was under the impression that these were mapped in, are they not present/not working?

It's not regular BS12 server. You don't have people to fix breaches or even care about them (due EVA suits - those people will just open everything on their way to reach a target without a mind about one who wear normal gear). Everyone are combantant. If you aren't one - you'll just get killed during main gameplay or get bored.

**Comment by Terror4000rus** [comment]
>Emergency firelocks are mapped in and work normally, they're on windows and doorways so whenever a breach happens it's limited to a (relatively) small area.

Marines and covenant constantly goes through airlocks and firedoors, which cause low atmosphere. It's not a fix for the problem since the main entertain thing in the server - combat. You just don't have people who care about low atmosphere, because EVA suits are in free access, but it's pain in ass to wear them. 

>Inevitably though, the entire ODP will be breached through assaulters maliciously opening doors (for combat advantage), creating new breaches (both for flanks and aforementioned combat advantage, there's a limited amount of EVA armour on the ODP) or both sides being forced to move through (thereby spreading the breach to other rooms) breached areas to reach an objective/threat/etc.

And that's what I'm talking about. I want EVA suits to be used **outside** the station. I don't like current mandatory state of them. You just suck if don't wear one during first phase of the game (ODP assault) and suck even more if the enemy doesn't make breaches.

**Comment by bloxgate** [comment]
I don't think getting rid of atmospheric effects is the solution here. People opening firelocks when there's no air on the other side, and not turning on magboots are all IC issues (assuming EVA gear comes with magboots, it's been a while since I looked at the EVA loadout).

We can certainly look into adding injection ports to the armor - they should have them given how often you need it.

**Comment by Terror4000rus** [comment]
> I don't think getting rid of atmospheric effects is the solution here. People opening firelocks when there's no air on the other side, and not turning on magboots are all IC issues (assuming EVA gear comes with magboots, it's been a while since I looked at the EVA loadout).

It's only solution to make ODP assault less painful for me. I don't wanna wear EVA suit every time I get on it. Marines even don't have enough of them. "It's IC issues - just turn on magboots, take oxygen tank and have fun" - somehow, it doesn't work. I actually like gameplay without atmosphere, at ground, but I don't like all the space it forces you to cover. Narrow station hallways make more sense for me.


> We can certainly look into adding injection ports to the armor - they should have them given how often you need it.

It's just a solution to don't loot medical vendor for hypospray. And even here, it's not fixing other problems.

**Comment by Sgtwok** [comment]
Increasing EVA suit number for both sides and including injection ports seems like a feasible solution to me. Cutting out an entire tactical function for Covenant to make the ODP manageable doesn't seem like a good one.

**Comment by caelaislinn** [comment]
Both Covenant and UNSC have a portable object for sealing breaches (vacuum shield and inflatable wall + airlock). Also, a recent PR (I can't find) made floors no longer breach to deep space or open space via explosions. I agree that the outer hull needs to be a bit thicker for some maps though

**Comment by caelaislinn** [comment]
#2771 added covenant organic shields which can be temporarily toggled, and also block air (map editor placement only). I'm going to close this because with the new cargo update, both factions can order their own temporary breach-sealer
